src/desktop/usb.rs

```diff
@@ -0,0 +1,326 @@
+#[zvariant(signature = "dict")]
+pub struct AcquireDevice {
+    writable: bool,
+}
+
+/// Device to acquire
+pub struct Device(String /* ID */, bool /* writable */);
+
+impl Device {
+    /// Create a new `Device`.
+    pub fn new(id: String, writable: bool) -> Device {
+        Device(id, writable)
+    }
+
+    /// Get the device id.
+    pub fn id(&self) -> &str {
+        &self.0
+    }
+
+    /// Get if the device is writable.
+    pub fn writable(&self) -> bool {
+        self.1
+    }
+}
+
+/// Option for AcquireDevice call.
+#[derive(SerializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+struct AcquireOptions {
+    handle_token: HandleToken,
+}
+
+/// Finished device acquired.
+#[derive(DeserializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+struct AcquiredDevice {
+    success: bool,
+    fd: Option<OwnedFd>,
+    error: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Type)]
+/// A usb event received part of the `device_event` signal response.
+pub struct UsbEvent(/* action */ String, /* id */ String, UsbDevice);
+
+impl UsbEvent {
+    /// The action.
+    pub fn event_action(&self) -> &str {
+        &self.0
+    }
+
+    /// The device id.
+    pub fn event_device_id(&self) -> &str {
+        &self.1
+    }
+
+    /// The [`UsbDevice`] properties.
+    pub fn event_device(&self) -> &UsbDevice {
+        &self.2
+    }
+}
+
+#[derive(Debug, Deserialize, Type)]
+/// A response received when the `device_event` signal is received.
+pub struct UsbDeviceEvent(OwnedObjectPath, Vec<UsbEvent>);
+
+impl UsbDeviceEvent {
+    /// The session triggered the state change
+    pub fn session_handle(&self) -> ObjectPath<'_> {
+        self.0.as_ref()
+    }
+
+    /// Events received
+    pub fn events(&self) -> &[UsbEvent] {
+        &self.1
+    }
+}
+
+/// This interface provides access to USB devices.
+#[derive(Debug)]
+#[doc(alias = "org.freedesktop.portal.Usb")]
+pub struct UsbProxy<'a>(Proxy<'a>);
+
+impl<'a> UsbProxy<'a> {
+    /// Create a new instance of [`UsbProxy`].
+    pub async fn new() -> Result<UsbProxy<'a>, Error> {
+        let proxy = Proxy::new_desktop("org.freedesktop.portal.Usb").await?;
+        Ok(Self(proxy))
+    }
+
+    /// Create a USB ession.
+    /// While this session is active, the caller will receive `DeviceEvents`
+    /// signals with device addition and removal.
+    ///
+    /// # Specifications
+    ///
+    /// See also [`CreateSession`](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Usb.html#org-freedesktop-portal-usb-createsession).
+    #[doc(alias = "CreateSession")]
+    pub async fn create_session(&self) -> Result<Session<'a, Self>, Error> {
+        let options = CreateSessionOptions::default();
+        let session: OwnedObjectPath = self.0.call("CreateSession", &(&options)).await?;
+        Session::new(session).await
+    }
+
+    /// Enumerate USB devices.
+    #[doc(alias = "EnumerateDevices")]
```

Comment:
This and the below methods lack links to the upstream docs, similar to what you have in create_session.

It would also be nice to describe what is returned in the various functions.